core/app/models/spree/tax/item_adjuster.rb

```diff
@@ -0,0 +1,32 @@
+module Spree
+  module Tax
+    class ItemAdjuster
+      include TaxHelpers
+
+      def initialize(item, options = {})
+        @item = item
+        @order = @item.order
+        # set the instance variable so `TaxRate.match` is only called when necessary
+        @order_rates = options[:order_tax_rates]
+      end
+
+      def adjust!
+        return unless order_tax_zone
+        # Using .destroy_all to make sure callbacks fire
+        @item.adjustments.tax.destroy_all
```

Comment:
after introduced we should use the `item` `attr_reader` here and not use instance variables.